linera-core/src/client/mod.rs

```diff
@@ -2008,7 +2014,12 @@ where
         // Create the final block proposal.
         let key_pair = self.key_pair().await?;
         let proposal = if let Some(cert) = manager.requested_locked {
-            Box::new(BlockProposal::new_retry(round, *cert, &key_pair, blobs))
+            Box::new(BlockProposal::new_retry(
+                round,
+                *cert,
+                &key_pair,
+                manager.locked_blobs,
```

Comment:
If we're gonna keep all used blobs in the locked blobs, we actually need to distinct here between published and used locked blobs 😅

Reply:
Right, good point, this should filter them and use only the published ones.